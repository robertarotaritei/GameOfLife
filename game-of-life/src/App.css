body, html, #root, .app, .container {
  width: 100%;
  height: 100%;
  background-color: #353535;
}


.container {
  font-size: 28px;
  color: #d9d9d9;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #353535;
}

.loginForm {
 box-sizing: border-box;
 width: 100%;
 max-width: 450px;
 padding-left: 20px;
 padding-right: 20px;
}

.inputField {
 padding-top: 16px;
}

.input {
 box-sizing: border-box;
 outline: none;
 border: solid 2px #3c6e71;
 border-radius: 4px;
 color: #292929;
 width: 100%;
 padding: 12px;
 font-size: 14px;
 background: rgba(255, 255, 255, 1);
}

.submitButton {
 padding-top: 16px;
}


.btn {
 width: 100%;
 min-width: 280px;
 color: #565656;
 padding: 12px;
 font-size: 14px;
 font-weight: bold;
 border: solid 2px #3c6e71;
 border-radius: 4px;
 background: #fff;
 cursor: pointer;
}

.topnav {
  background-color: #333;
  overflow: hidden;
}

.topnav a {
  float: left;
  color: #d9d9d9;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 17px;
}

.topnav a:hover {
  background-color: #d9d9d9;
  color: #d9d9d9;
}

.topnav a.active {
  background-color: #4CAF50;
  color: white;
}

.linkbutton {
  align-items: normal;
  background-color: #353535;
  border-color: #d9d9d9;
  border-style: none;
  box-sizing: content-box;
  color: #d9d9d9;
  cursor: pointer;
  display: inline;
  font-size: 28px;
  padding: 0;
  perspective-origin: 0 0;
  text-align: start;
  text-decoration: underline;
  transform-origin: 0 0;
  -moz-appearance: none;
  -webkit-logical-height: 1em;
  -webkit-logical-width: auto;
}