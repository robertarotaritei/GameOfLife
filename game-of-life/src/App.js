import React from 'react';
import { observer } from 'mobx-react'
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import './App.css'
import UserStore from './stores/UserStore';
import LoginForm from './components/Login and Registration/LoginForm';
import RegistrationForm from './components/Login and Registration/RegistrationForm';
import Dashboard from './components/pages/Dashboard';
import Welcome from './components/pages/Welcome';
import { runInAction } from 'mobx';
import Typography from '@material-ui/core/Typography';

class App extends React.Component {

  constructor() {
    super();

    runInAction(() => {
      UserStore.key = sessionStorage.getItem('key');
      UserStore.username = sessionStorage.getItem('username');
      UserStore.loading = false;
    });
  }

  async componentDidMount() {
    let result = await fetch(`/credentials/user/token?username=${UserStore.username}&token=${UserStore.key}`);
    result = await result.json();

    if (result !== true) {
      sessionStorage.removeItem('key');
      sessionStorage.removeItem('username');
      sessionStorage.removeItem('selectedPage');
      sessionStorage.removeItem('initialState');
      runInAction(() => {
        UserStore.key = "";
        UserStore.username = "";
      });
    }
  }

  render() {
    if (UserStore.loading) {
      return (
          <div className="app">
            <div className='container'>
              <Typography variant='h3' style={{ fontWeight: 'bold', textAlign: 'center' }}>
                Loading, please wait...
            </Typography>
            </div>
          </div>
      )
    }
    else {

      if (UserStore.key) {
        return (
            <div className="app">
              <Router>
                <Switch>
                  <Route exact path="/" component={Dashboard} />
                  <Route exact path="/*" render={() => (<Redirect to="/" />)} />
                </Switch>
              </Router>
            </div>
        );
      }

      return (
          <div className="app">
            <div className="container">
              <Router>
                <Switch>
                  <Route exact path="/" component={Welcome} />
                  <Route exact path="/welcome" component={Welcome} />
                  <Route exact path="/register" component={RegistrationForm} />
                  <Route exact path="/login" component={LoginForm} />
                  <Route exact path="/*" render={() => (<Redirect to="/login" />)} />
                </Switch>
              </Router>
            </div>
          </div>
      );
    }
  }
}

export default observer(App);