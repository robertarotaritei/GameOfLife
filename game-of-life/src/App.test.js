import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import { shallow } from "enzyme";

test("App renders without crashing", () => {
  shallow(<App />);
});

test('Welcome Page shows up', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Welcome to the Game of Life/i);
  expect(linkElement).toBeInTheDocument();
});

it('App renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});