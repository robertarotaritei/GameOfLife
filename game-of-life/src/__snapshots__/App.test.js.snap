// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <div
      className="app"
    >
      <div
        className="container"
      >
        Welcome to the Game of Life.
        <br />
        <br />
        Click here to log in 
        <button
          onClick={[Function]}
        >
          Log in
        </button>
        <br />
        <br />
        Or here if you don't have an account already 
        <button
          onClick={[Function]}
        >
          Register
        </button>
      </div>
    </div>
  </div>
</div>
`;
